src/Paginator.php

```diff
@@ -894,7 +894,7 @@ protected function _getCacheId($page = null)
     protected function _getCacheInternalId()
     {
         $adapter            = $this->getAdapter();
-        $adapterToSerialize = $adapter instanceof DbSelect
+        $adapterToSerialize = $adapter instanceof ArrayCopyInterface
```

Comment:
This part is already deprecated and must be removed in the next major version. Therefore I would simplify the whole thing and reduce it to a bugfix: `method_exists()`

Reply:
I though about method_exists but I also though it was not a goo practice ^^"
But if you are ok with that, I can simplify the code

Reply:
Since this is a bugfix on top of broken code, some ugliness is kinda needed.

The added test will make up for the ugly conditional.

Reply:
I'm struggling on how to test that ..... will come back soon :D

Reply:
@e-belair 
It is very commendable that you prefer a clean solution, but at this point the test is more important than the implementation.